src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala

```diff
@@ -191,6 +279,14 @@ object XSDToSchema {
     }
   }
 
+  private def getBaseStructField(xmlSchema: XmlSchema, schemaType: XmlSchemaType): StructField = {
+    if (schemaNameToStructFieldMap.contains(schemaType.getName)){
+      return schemaNameToStructFieldMap.getOrElse(schemaType.getName, null);
```

Comment:
return is not needed in Scala
You already know the map has this value. You can do this all in one getOrElse